pkg/resource/obtenantrestore_manager.go

```diff
@@ -72,11 +77,77 @@ func (m ObTenantRestoreManager) FinishTask() {
 }
 
 func (m ObTenantRestoreManager) HandleFailure() {
+	if m.IsDeleting() {
+		m.Resource.Status.OperationContext = nil
+	} else {
+		operationContext := m.Resource.Status.OperationContext
+		failureRule := operationContext.OnFailure
+		switch failureRule.Strategy {
+		case "":
+			fallthrough
+		case strategy.StartOver:
+			m.Resource.Status.Status = constants.RestoreJobStatus(failureRule.NextTryStatus)
+			m.Resource.Status.OperationContext = nil
+		case strategy.RetryFromCurrent:
+			operationContext.TaskStatus = taskstatus.Pending
+		case strategy.Pause:
+		}
+	}
+}
 
+func (m *ObTenantRestoreManager) checkRestoreProgress() error {
+	con, err := m.getOperationManager()
+	if err != nil {
+		return err
+	}
+	restoreJob, err := con.GetLatestRestoreProgressOfTenant(m.Resource.Spec.TargetTenant)
+	if err != nil {
+		return err
+	}
+	if restoreJob != nil {
+		m.Resource.Status.RestoreProgress = &model.RestoreHistory{RestoreProgress: *restoreJob}
+		if restoreJob.Status == "SUCCESS" {
+			m.Recorder.Event(m.Resource, corev1.EventTypeNormal, "Restore job finished", "Restore job finished")
+			if m.Resource.Spec.RestoreRole == constants.TenantRoleStandby {
+				m.Resource.Status.Status = constants.RestoreJobStatusReplaying
+			} else {
+				m.Resource.Status.Status = constants.RestoreJobStatusActivating
+			}
+		} else if restoreJob.Status == "FAIL" {
+			m.Recorder.Event(m.Resource, corev1.EventTypeWarning, "Restore job is failed", "Restore job is failed")
+			m.Resource.Status.Status = constants.RestoreJobFailed
+		}
+	}
+	if restoreJob == nil {
```

Comment:
simply use else

Reply:
Actually I don't remember I write such block. Maybe Github copilot generate the condition hahah.